pyproject.toml

```diff
@@ -100,6 +100,20 @@ features = [
   "dev-packages",
 ]
 
+[tool.hatch.envs.postgres-env]
+description = "Dev environment for working with Postgres adapter"
+pre-install-commands = [
+  "pip install dbt-postgres",
```

Comment:
What's the reason for installing via pip instead of specifying it as a dependency?

Reply:
I could've sworn I had a comment about this, but clearly there isn't one. I'll add one before merging.

We don't depend on dbt-postgres, or any specific adapter, anywhere in the codebase, so we don't want to force an install in the normal dependency list. We could make it an env-specific dependency but then we're bound to a version range, which means keeping the version ranges in sync between here and where we specify dbt-core, and dealing with things like dbt-semantic-interfaces version requirements.

This way we get the relevant adapter installed in the environment, the way an end user would, and pip takes care of the rest as best it's able.

Once we have all of the packaging laid out with dbt-metricflow we'll likely update this to use an editable dependency on the local package. The pre-install is apparently the only way hatch supports editable dependency installations, so that'll have to be here anyway, but that should allow us to work around dbt-semantic-interfaces version update conflicts in most circumstances.